js/src/Ice/ConnectionI.js

```diff
@@ -1140,42 +1144,43 @@ export class ConnectionI {
         return true;
     }
 
+    /**
+     * Called when the connection is ready to start sending the next message in the send queue. The connection queues
+     * protocol messages to be sent when the sending cannot complete synchronously or when another protocol message is
+     * in the process of being sent.
+     *
+     * When this method is called, the first message in the send queue (the message currently being sent) has already
+     * been sent and will be removed from the queue.
+     *
+     * This method will continue sending messages until the send queue is empty or until a protocol message cannot be
+     * sent synchronously.
+     *
+     * If the first message in the queue is a protocol request and its response was received before this method was
+     * called, the response should be cached in the message's response field and will be returned as the result. The
+     * caller is then responsible for processing the response, now that the message has been marked as sent.
+     *
+     * @returns The response of the first message in the send queue if it has been received; null otherwise.
+     */
     sendNextMessage() {
-        if (this._sendStreams.length === 0) {
```

Comment:
This cannot happen in JavaScript implementation, `sendNextMessage` it is only called when we are sending messages.